include/seqan3/range/view/take_until.hpp

```diff
@@ -0,0 +1,453 @@
+    constexpr view_take_until & operator=(view_take_until const & rhs) = default;
+    constexpr view_take_until & operator=(view_take_until && rhs) = default;
+    ~view_take_until() = default;
+
+    /*!\brief Construct from another range.
+     * \param[in] _urange The underlying range.
+     * \param[in] _fun    The functor that acts as termination criterium.
+     */
+    view_take_until(urng_t _urange, fun_t _fun)
+        : urange{std::move(_urange)}, fun{std::forward<fun_t>(_fun)}
+    {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the container.
+     * \returns Iterator to the first element.
+     *
+     * If the container is empty, the returned iterator will be equal to seqan3::view_take_until::end().
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() noexcept
+    {
+        return {ranges::begin(urange), static_cast<fun_t &>(fun)};
+    }
+
+    //!\copydoc begin()
+    const_iterator begin() const noexcept
+        requires const_iterable
+    {
+        return {ranges::cbegin(urange), static_cast<fun_t const &>(fun)};
+    }
+
+    //!\copydoc begin()
+    const_iterator cbegin() const noexcept
+        requires const_iterable
+    {
+        return {ranges::cbegin(urange), static_cast<fun_t const &>(fun)};
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the range.
+     * \returns Iterator to the end.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    sentinel_type end() noexcept
+    {
+        return {ranges::end(urange)};
+    }
+
+    //!\copydoc end()
+    sentinel_type end() const noexcept
+        requires const_iterable
+    {
+        return {ranges::cend(urange)};
+    }
+
+    //!\copydoc end()
+    sentinel_type cend() const noexcept
+        requires const_iterable
+    {
+        return {ranges::cend(urange)};
+    }
+    //!\}
+
+    /*!\brief Convert this view into a container implicitly.
+     * \tparam container_t Type of the container to convert to; must model seqan3::container_concept and it's
+     *                     seqan3::reference_t must model seqan3::common_reference_concept with `reference`.
+     * \returns This view converted to container_t.
+     */
+    template <container_concept container_t>
+    operator container_t()
+    //!\cond
+        requires common_reference_concept<reference_t<container_t>, reference>
+    //!\endcond
+    {
+        container_t ret;
+        ranges::copy(begin(), end(), ranges::back_inserter(ret));
+        return ret;
+    }
+};
+
+//!\brief Type deduction guide that strips references.
+//!\relates seqan3::detail::view_take_until
```

Comment:
multiline comment with `/**/`

Reply:
For two lines it's ok I would argue because the the overhead for `/**/` is 50% of the entire number of lines :stuck_out_tongue: